plugins/c8y_configuration_plugin/Cargo.toml

```diff
@@ -34,6 +34,7 @@ assets = [
 anyhow = "1.0"
 c8y_config_manager = { path = "../../crates/extensions/c8y_config_manager" }
 c8y_http_proxy = { path = "../../crates/extensions/c8y_http_proxy" }
+certificate = { path = "../../crates/common/certificate" }
```

Comment:
Adding dependency on `certificate` crate just for `CertificateError` is not a good idea. Add `CertificateError` to `lib.rs` of `tedge_config` crate.

So, add this line to `crates/common/tedge_config/src/lib.rs`.

```rust
pub use certificate::CertificateError;
```

Similar example. When you implement something using `Topic`, you just need to add a dependency on `tedge_mqtt_ext` and include `tedge_mqtt_ext::Topic` instead of `mqtt_channel::Topic`, although the actual definition of `Topic` is inside of `mqtt_channel` crate.